atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/table/common/TableTasks.java

```diff
@@ -90,29 +94,26 @@ public static void copyExternal(ExecutorService exec,
                                     int threadCount,
                                     final CopyStats stats,
                                     final CopyTask task) throws InterruptedException {
-        BlockingWorkerPool pool = new BlockingWorkerPool(exec, threadCount);
-        for (final MutableRange range : getRanges(threadCount, batchSize)) {
-            pool.submitTask(() -> {
-                do {
-                    final RangeRequest request = range.getRangeRequest();
-                    try {
-                        long startTime = System.currentTimeMillis();
-                        PartialCopyStats partialStats = task.call(request, range);
-                        stats.rowsCopied.addAndGet(partialStats.rowsCopied);
-                        stats.cellsCopied.addAndGet(partialStats.cellsCopied);
-                        log.info("Copied {} rows, {} cells from {} to {} in {} ms.",
-                                partialStats.rowsCopied,
-                                partialStats.cellsCopied,
-                                srcTable,
-                                dstTable,
-                                System.currentTimeMillis() - startTime);
-                    } catch (InterruptedException e) {
-                        throw Throwables.rewrapAndThrowUncheckedException(e);
-                    }
-                } while (!range.isComplete());
-            });
-        }
-        pool.waitForSubmittedTasks();
+        new InterruptibleRangeExecutor(exec, batchSize, threadCount).executeTask("copy",
+                range -> executeCopyTask(srcTable, dstTable, stats, task, range));
+    }
+
+    private static void executeCopyTask(TableReference srcTable,
+            TableReference dstTable,
+            CopyStats stats,
+            CopyTask task,
+            MutableRange range) throws InterruptedException {
+        final RangeRequest request = range.getRangeRequest();
```

Comment:
nit: This doesn't seem like it needs to be final any more, since we're doing the "magic" in `task.call()`